cmd/ctl/go.mod

```diff
@@ -2,13 +2,11 @@ module github.com/cert-manager/cert-manager/cmctl-binary
 
 go 1.20
 
-replace github.com/cert-manager/cert-manager => ../../
-
 // remove this once controller-runtime v0.15.0 is released
 replace sigs.k8s.io/controller-runtime v0.14.6 => sigs.k8s.io/controller-runtime v0.13.1-0.20230503134813-2e57de78ba00
 
 require (
-	github.com/cert-manager/cert-manager v0.0.0-00010101000000-000000000000
+	github.com/cert-manager/cert-manager v1.12.0-beta.1
```

Comment:
So basically if we want to release with a cmctl pointing to an up-to-date cert-manager version then we should bump this version to the latest cert-manager commit SHA that changes cert-manager code.
Suggestion: should we add a comment here about this so it's clearer?

Reply:
A comment sounds good, I'll add one.

To be fair, maybe using a tag here is gonna confuse things since we're probably more likely to use a commit SHA. Maybe I should use a SHA here.

Reply:
Done!